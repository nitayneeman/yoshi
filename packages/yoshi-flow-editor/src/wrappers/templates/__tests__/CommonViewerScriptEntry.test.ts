import commonViewerScriptEntry from '../CommonViewerScriptEntry';

describe('CommonViewerScriptEntry template', () => {
  it('generates correct template with single controller', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          id: '123',
        },
      ],
      initAppPath: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with multiple controllers', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '123',
        },
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '567',
        },
      ],
      initAppPath: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template w/o controllers', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      controllersMeta: [],
      initAppPath: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });
});
