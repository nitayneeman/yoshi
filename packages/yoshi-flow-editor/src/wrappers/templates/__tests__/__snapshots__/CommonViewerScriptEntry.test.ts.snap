// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonViewerScriptEntry template generates correct template w/o controllers 1`] = `
"
  import {createControllersWithDescriptors, initAppForPage as initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';

  import userInitApp from 'project/src/app.ts';
  
  


  export const initAppForPage = initAppForPageWrapper;
  export const createControllers = createControllersWithDescriptors([], userInitApp);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with multiple controllers 1`] = `
"
  import {createControllersWithDescriptors, initAppForPage as initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';

  import userInitApp from 'project/src/app.ts';
  
  import controller0 from 'project/src/components/todo/controller.ts';
import controller1 from 'project/src/components/todo/controller.ts';


  export const initAppForPage = initAppForPageWrapper;
  export const createControllers = createControllersWithDescriptors([{ method: controller0, id: \\"123\\" }, { method: controller1, id: \\"567\\" }], userInitApp);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with single controller 1`] = `
"
  import {createControllersWithDescriptors, initAppForPage as initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';

  import userInitApp from 'project/src/app.ts';
  
  import controller0 from 'project/src/components/button/controller.ts';


  export const initAppForPage = initAppForPageWrapper;
  export const createControllers = createControllersWithDescriptors([{ method: controller0, id: \\"123\\" }], userInitApp);
"
`;
